### PR TITLE
User/bt/linux build improvements

### PR DESCRIPTION
**Pull request by Ingrater**
- Add support for using the Ninja generator when building for linux
- Add support for building with clang when building for linux
- Give error message when uuid.h is missing
- Give better error message when the wrong CMAKE_BUILD_TYPE is used (which can easily happen on the command line)

### CODE REVIEW

File: Code/BuildSystem/CMake/ezUtils.cmake [hunk 1]
```diff
@@ -65,7 +67,7 @@ function(ez_set_target_output_dirs TARGET_NAME LIB_OUTPUT_DIR DLL_OUTPUT_DIR)
 			ARCHIVE_OUTPUT_DIRECTORY "${OUTPUT_LIB_DEV}"
         )
     else()
-        message(WARNING "Unknown CMAKE_BUILD_TYPE: '${CMAKE_BUILD_TYPE}'")
+        message(WARNING "Unknown CMAKE_BUILD_TYPE: '${CMAKE_BUILD_TYPE}'. ezEngine cmake scripts support the following 3 build types: 'Debug', 'Dev', 'Release'")
```
Comment (jankrassnigg) [inline Code/BuildSystem/CMake/ezUtils.cmake]: "Shipping" not "Release" IIRC

File: Code/BuildSystem/CMake/ezUtils.cmake [hunk 2]
```diff
@@ -245,8 +247,8 @@ function(ez_set_library_properties TARGET_NAME)
     endif ()
 
     if (EZ_CMAKE_PLATFORM_OSX OR EZ_CMAKE_PLATFORM_LINUX)
-        find_package(X11 REQUIRED)
-        find_package(SFML REQUIRED system window)
+		find_package(X11 REQUIRED)
+		find_package(SFML REQUIRED system window)
         target_include_directories (${TARGET_NAME} PRIVATE ${X11_X11_INCLUDE_PATH})
         target_link_libraries (${TARGET_NAME} PRIVATE ${X11_X11_LIB} sfml-window sfml-system)
```
Comment (jankrassnigg) [inline Code/BuildSystem/CMake/ezUtils.cmake]: nit: formatting is somehow inconsistent

File: Code/Engine/Foundation/CMakeLists.txt [hunk 3]
```diff
@@ -18,6 +18,11 @@ if (MSVC)
 endif()
 
 if (EZ_CMAKE_PLATFORM_LINUX)
+  CHECK_INCLUDE_FILE_CXX("uuid/uuid.h" UUID_HEADER)
+  if (NOT UUID_HEADER)
+	message(FATAL_ERROR "Missing uuid/uuid.h. Have you installed the uuid-dev package?")
+  endif()
```
Comment (jankrassnigg) [inline Code/Engine/Foundation/CMakeLists.txt]: nit: formatting

**Review by jankrassnigg — APPROVED** [review]

**Comment by Ingrater** [comment]
@Sanakan8472 the WindowsUWP builds keep failing. Seems to be unrelated. Can you take a look?

**Comment by jankrassnigg** [comment]
> @Sanakan8472 the WindowsUWP builds keep failing. Seems to be unrelated. Can you take a look?

That's just a vs license renewal thing. You can ignore that.

**Pull request merged**